### PR TITLE
Add integration test tools for MORYX modules

### PR DESCRIPTION
**Pull request by 1nf0rmagician**
Adding a `MoryxTestEnvironment` as well as a tutorial on how to use it for integration testing of modules.

### CODE REVIEW

**Comment by Toxantron** [comment]
I removed the class generic as it makes things unncessarily complicated. It was only used to extract the module from the service collection via its facade registration. Since modules are [registered under their type](https://github.com/PHOENIXCONTACT/MORYX-Framework/blob/dev/src/Moryx.Runtime.Kernel/KernelServiceCollectionExtensions.cs#L57) this could be simplified.

**Comment by 1nf0rmagician** [comment]
I like it, adjusted the tutorial to match the non generig class

File: docs/tutorials/HowToTestAModule.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Setup a test environment for integration tests of a module
+
+In order to test a module in its lifecycle with its respective facade we offer the `Moryx.TestTools.IntegrationTest`.
+The package brings a `MoryxTestEnvironment<T>`.
+With this class you can first create mocks for all module facades your module dependents on using the static `CreateModuleMock<FacadeType>` method.
```
Comment (seveneleven) [inline docs/tutorials/HowToTestAModule.md:5]: Not in sync with the code?

File: src/Moryx.TestTools.IntegrationTest/MoryxTestEnvironment.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    /// </summary>
+    /// <typeparam name="T">Type of the facade to be tested.</typeparam>
+    public class MoryxTestEnvironment
+    {
+        private readonly Type _moduleType;
+
+        public IServiceProvider Services { get; private set; }
+
+        /// <summary>
+        /// Creates an <see cref="IServiceProvider"/> for integration tests of moryx. We prepare the
```
Comment (seveneleven) [inline src/Moryx.TestTools.IntegrationTest/MoryxTestEnvironment.cs:33]: 'It' prepares?

**Review by Toxantron — APPROVED** [review]

**Review by seveneleven — COMMENTED** [review]
I like it!

**Pull request merged**